### PR TITLE
[front] Rework of LocalStorage

### PR DESCRIPTION
**Pull request by Kedae**

### CODE REVIEW

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx [hunk 1]
```diff
@@ -3,6 +3,8 @@ export const FiltersVariant = {
   dialog: 'dialog',
 };
 
+export type BackendFilters = { key: string, values: string[], operator: string, filterMode: string }[];
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx:6]: It's only use for the backend and never show to the user ?

**Review by RomuDeuxfois — APPROVED** [review]

File: opencti-platform/opencti-front/src/private/components/analysis/Reports.tsx [hunk 2]
```diff
@@ -63,45 +80,6 @@ const Reports: FunctionComponent<ReportsProps> = ({
...
   );
 
-  const { reportType } = match.params;
-
-  const reportFilterClass = reportType !== 'all' && reportType !== undefined
-    ? reportType.replace(/_/g, ' ')
-    : '';
-
-  const additionnalFilters = [];
-  if (reportFilterClass) {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/analysis/Reports.tsx:73]: Why are you removing that ?

**Review by RomuDeuxfois — APPROVED** [review]

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx (hunk 1 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx:6]: It's not about what is shown or not: it's the format. We don't have the same filter format for both front and back. I think there is a figma about that, you can ask Cathia.

File: opencti-platform/opencti-front/src/private/components/analysis/Reports.tsx (hunk 2 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/analysis/Reports.tsx:73]: I only removed the reportType part, the additionalFilters are above now.
The reportType is useless as it is part of the URL and filters and that's exactly what useLocalStorage is designed for, so it's redundant.

**Pull request merged**